src/Microsoft.Dnx.Tooling/Publish/PublishManager.cs

```diff
@@ -58,8 +58,18 @@ public bool Publish()
                 _options.Reports.Information.WriteLine(warning.FormattedMessage.Yellow());
             }
 
-            // '--wwwroot' option can override 'webroot' property in project.json
-            _options.WwwRoot = _options.WwwRoot ?? project.WebRoot;
+            if (string.IsNullOrEmpty(_options.WwwRoot) && File.Exists(Path.Combine(project.ProjectDirectory, "Microsoft.AspNet.Hosting.json")))
+            {
+                var jsonObj = JObject.Parse(File.ReadAllText(Path.Combine(project.ProjectDirectory, "Microsoft.AspNet.Hosting.json")));
+                _options.WwwRoot = jsonObj?["Hosting"]?["WebRoot"]?.ToString();
```

Comment:
use the config APIs, don't read the json directly. There are variable layouts like `"Hosting:WebRoot": "wwwroot"`